src/content/docs/en/guides/actions.mdx

```diff
@@ -560,24 +550,155 @@ const result = Astro.getActionResult(actions.addToCart);
+  // Skip requests for prerendered pages
+  if (context.isPrerendered) return next();
+  
+  const { action, setActionResult, serializeActionResult } =
+    getActionContext(context);
+  // Create a Blob store to persist action results with Netlify Blob
+  const actionStore = getStore("action-session");
+  
+  // If an action result was forwarded as a cookie, set the result
+  // to be accessible from `Astro.getActionResult()`
+  const sessionId = context.cookies.get("action-session-id")?.value;
+  const session = sessionId
+    ? await actionStore.get(sessionId, {
+        type: "json",
+      })
+    : undefined;
+  
+  if (session) {
+    setActionResult(session.actionName, session.actionResult);
+  
+    // Optional: delete the session after the page is rendered.
+    // Feel free to implement your own persistence strategy
+    await actionStore.delete(sessionId);
+    ctx.cookies.delete("action-session-id");
+    return next();
+  }
+  
+  // If an action was called from an HTML form action,
+  // call the action handler and redirect to the destination page
+  if (action?.calledFrom === "form") {
+    const actionResult = await action.handler();
+  
+    // Persist the action result using session storage
+    const sessionId = randomUUID();
+    await actionStore.setJSON(sessionId, {
+      actionName: action.name,
+      actionResult: serializeActionResult(actionResult),
+    });
+  
+    // Pass the session ID as a cookie
+    // to be retrieved after redirecting to the page
+    context.cookies.set("action-session-id", sessionId);
+  
+    // Redirect back to the previous page on error
+    if (actionResult.error) {
+      const referer = context.request.headers.get("Referer");
+      if (!referer) {
+        throw new Error(
+          "Internal: Referer unexpectedly missing from Action POST request.",
+        );
+      }
+      return context.redirect(referer);
+    }
+    // Redirect to the destination page on success
+    return context.redirect(context.originPathname);
+  }
+  
+  return next();
+});
+```
+
+## Security when using actions
+
+Actions are accessible as public endpoints based on the name of the action. For example, the action `blog.like()` will be accessible from `/_actions/blog.like`. This is useful for unit testing action results and debugging production errors. However, this means you **must** use same authorization checks that you would consider for API endpoints and on-demand rendered pages.
+
+### Authorize users from an action handler
+
+The easiest way to authorize action requests is to add an authentication check to your action handler. You may reach for [an authentication library](https://docs.astro.build/en/guides/authentication/) to handle session management and user information. Actions expose the full `APIContext` object to access properties passed from middleware using `context.locals`. When a user is not authorized, you can raise an `ActionError` with the `UNAUTHORIZED` code:
```

Comment:
```suggestion
To authorize action requests, add an authentication check to your action handler. You may want to use [an authentication library](https://docs.astro.build/en/guides/authentication/) to handle session management and user information.

Actions expose the full `APIContext` object to access properties passed from middleware using `context.locals`. When a user is not authorized, you can raise an `ActionError` with the `UNAUTHORIZED` code:
```
Just felt like it wanted to be a new paragraph, I think? Also:
- don't tell people what is easy (and, it's ok to only tell them the happy path, even if there are more ways)
- I think something a little more bland than "reach for" is fitting for docs: "you may choose to use", "you may want to use", "you can use... or/instead of building your own authentication" Pick what you like, but less imagery, I think!

Reply:
Same don't be fooled by outdated here!